### PR TITLE
Add tests for `time_cell_min` and update documentation for the parameter

### PR DESCRIPTION
**Pull request by freemansw1**
Follows from my mistake in #360 - this adds a specific test for time_cell_min and updates the documentation for the parameter in the tracking API docs to reflect the correct units (not minutes). 

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [x] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [x] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Review by JuliaKukulies — APPROVED** [review]
Approved, but a question to double-check:  Can `dt` really be provided in any different units than seconds? 

I am thinking of this line here where `dt` is related to `vmax`:
https://github.com/tobac-project/tobac/blob/09b5a733e0309d8dde47d7ceeccd273cdc8aa7dd/tobac/tracking.py#L234 

Maybe we should also add in the documentation then that `vmax` needs to be per same time unit as `dt`. Same is true for `dxy`

**Comment by freemansw1** [comment]
> Approved, but a question to double-check: Can `dt` really be provided in any different units than seconds?
> 
> I am thinking of this line here where `dt` is related to `vmax`:
> 
> https://github.com/tobac-project/tobac/blob/09b5a733e0309d8dde47d7ceeccd273cdc8aa7dd/tobac/tracking.py#L234
> 
> Maybe we should also add in the documentation then that `vmax` needs to be per same time unit as `dt`. Same is true for `dxy`

This is a good point- these units all need to match (and communicating that without specifying units can be a headache). I wonder if we shouldn't just require that `dt` be in seconds, `dxy` in meters, and `vmax` in m/s? In theory these can all be in different units as long as they are consistent, but may be easier just to message the proper units.

**Comment by JuliaKukulies** [comment]
> > Approved, but a question to double-check: Can `dt` really be provided in any different units than seconds?
> > I am thinking of this line here where `dt` is related to `vmax`:
> > https://github.com/tobac-project/tobac/blob/09b5a733e0309d8dde47d7ceeccd273cdc8aa7dd/tobac/tracking.py#L234
> > 
> > Maybe we should also add in the documentation then that `vmax` needs to be per same time unit as `dt`. Same is true for `dxy`
> 
> This is a good point- these units all need to match (and communicating that without specifying units can be a headache). I wonder if we shouldn't just require that `dt` be in seconds, `dxy` in meters, and `vmax` in m/s? In theory these can all be in different units as long as they are consistent, but may be easier just to message the proper units.

Yes, totally agree that it is a bit difficult to communicate this and that it therefore would be best to just stick to SI units

**Comment by JuliaKukulies** [comment]
We discussed this with the whole developer group and agreed that it would be easiest to require SI units for all input parameters, i.e. seconds for times, meters for distances, m/s for speeds etc. In the future, it would maybe be smart to test for consistent units as well, but I think we will be OK for now with just updating the docstrings. Did you want a second reviewer for this as well @freemansw1 ?

**Comment by freemansw1** [comment]
> We discussed this with the whole developer group and agreed that it would be easiest to require SI units for all input parameters, i.e. seconds for times, meters for distances, m/s for speeds etc. In the future, it would maybe be smart to test for consistent units as well, but I think we will be OK for now with just updating the docstrings. Did you want a second reviewer for this as well @freemansw1 ?

Yes, good call! Checking units can be a bit of a headache (we could go down the path of requiring units like metpy does, but I've found dealing with pint et al. to be cumbersome), but worth exploring for the future. For now, we can just document the required units.

File: tobac/tests/test_tracking.py
Comment (snilsn) [inline tobac/tests/test_tracking.py:612]: Why go through the struggle of this line and have to use floor division twice, when you could just set `num_frames` to the original cell time lengths (so that `delta_time` would be 1, no division needed) and then assert against the original cell time lengths again at the end?

**Review by snilsn — APPROVED** [review]
I'm convinced this does what it should, so I'm approving it. I only have one question/comment regarding readability, but I guess that doesn't really matter since it's just a test.

File: tobac/tests/test_tracking.py
Comment (freemansw1) [reply tobac/tests/test_tracking.py]: Good point. I've updated the code just to set `delta_time` to 1.

**Comment by freemansw1** [comment]
Ok, I've addressed @snilsn's comment and have gone through and specified units in the tracking documentation. @JuliaKukulies Can you check that before we merge?

**Comment by JuliaKukulies** [comment]
> Ok, I've addressed @snilsn's comment and have gone through and specified units in the tracking documentation. @JuliaKukulies Can you check that before we merge?

Looks good, thanks for adding the units for the tracking documentation! 

Can we add the same for `dxy` and `dz` in the feature detection and segmentation? I could not directly push commits to this PR, otherwise I would have added them myself

**Comment by freemansw1** [comment]
> > Ok, I've addressed @snilsn's comment and have gone through and specified units in the tracking documentation. @JuliaKukulies Can you check that before we merge?
> 
> Looks good, thanks for adding the units for the tracking documentation!
> 
> Can we add the same for `dxy` and `dz` in the feature detection and segmentation? I could not directly push commits to this PR, otherwise I would have added them myself

Good point. I've gone through and made updates to the documentation of both segmentation and feature detection.

**Pull request merged**
